src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -56,9 +56,37 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             catch
             {
                 string message = String.Format(CultureInfo.CurrentCulture,
-                    "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                    "Invalid storage account '{0}'. Please make sure your credentials are correct.",
+                    account.Credentials.AccountName);
                 throw new InvalidOperationException(message);
             }
+
+            if (isPrimaryAccount)
+            {
+                // Primary storage accounts require Queues
+                var queueClient = account.CreateQueueClient();
+                var queue = queueClient.GetQueueReference("name");
+                try
+                {
+                    await queue.ExistsAsync(cancellationToken);
+                    _primaryCredentials = account.Credentials;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (StorageException exception)
+                {
+                    if (exception.Message.Contains("The remote name could not be resolved"))
+                    {
+                        string message = String.Format(CultureInfo.CurrentCulture,
+                            "Invalid storage account '{0}'. Primary storage accounts must be general "
+                            + "purpose accounts and not restricted blob storage accounts.", account.Credentials.AccountName);
```

Comment:
Invalid hosts are caught with this call to the blob endpoint: https://github.com/Azure/azure-webjobs-sdk/pull/827/files#diff-d66d3cb7ae4b38012579fd160bda4f73R50

We only reach this point if the blob endpoint call is successful, indicating that credentials work, but there is no queue endpoint.